### PR TITLE
spreg Documentation website

### PR DESCRIPTION
**Pull request by weikang9009**
This PR is to 
* complete and publish the documentation website with readthedocs. I have tested it on my  fork and it builds successfully https://spreg-wei.readthedocs.io/en/latest/

* configure travis CI to for dual testing dependency libpysal on pypi and github (allow failure for the most updated github version). The travis error seems to come from `libpysal` weight's `silent_island_warning` as this attribute was renamed to `silence_warnings` but not all occurrences were updated (also reported in https://github.com/pysal/libpysal/issues/204)
image

### CODE REVIEW

**Comment by weikang9009** [comment]
Not sure why we still run into the weight silence warning error after the issue is resolved in libpysal https://github.com/pysal/libpysal/pull/206. Testing against the libpysal github version should not give such error https://travis-ci.org/pysal/spreg/jobs/616823150?utm_medium=notification&utm_source=email any more. @ljwolf @jGaboardi could you please take a look at this when you have a chance?

**Comment by jGaboardi** [comment]
Yeah, something's happening weird consider [the problem should be solved](https://github.com/pysal/libpysal/issues/204#issuecomment-558317253)...

File: .travis.yml [hunk 1]
```diff
@@ -29,18 +34,22 @@ before_install:
...
   - source activate test-env
 
 install:
   - conda install --yes pip
   - if "$PYSAL_PYPI"; then
         echo 'testing pypi libpysal' && pip install libpysal;
-        else echo 'testing git libpysal'; git clone https://github.com/pysal/libpysal.git; cd libpysal; pip install .; cd ../;
+    else
+        echo 'testing git libpysal';
+        pip install https://github.com/pysal/libpysal/archive/master.zip;
```
Comment (jGaboardi) [inline .travis.yml]: maybe this line is the culprit? are we grabbing from the archived master here? [this is the corresponding line for `spaghetti`](https://github.com/pysal/spaghetti/blob/68b536f387f8c0ce5e70566b6aaa52ff49b3d2b4/.travis.yml#L64)

Comment (weikang9009) [reply .travis.yml]: It should be fine - `splot` and `giddy` are using the same strategy. https://github.com/pysal/libpysal/archive/master.zip grabs the current version of libpysal on the `master` branch.

Comment (jGaboardi) [reply .travis.yml]: hmmm. I will try to make some time to give this attention in the next few days.

**Comment by jGaboardi** [comment]
@weikang9009 check out [this](https://travis-ci.org/pysal/spreg/jobs/616839855#L384) and [this](https://travis-ci.org/pysal/spreg/jobs/616839855#L535).

Seems like the `libpysal` requirement from within [`requirements.txt`](https://github.com/pysal/spreg/blob/master/requirements.txt) is leading to it being installed from `pypi` (the current tagged release) and then being passed over when we are asking for the current master.

**Comment by jGaboardi** [comment]
In `spaghetti`, [`libpysal` is already installed, but by cloning the `libpysal` repo the install is updated to the current master](https://travis-ci.org/pysal/spaghetti/jobs/616964788#L764-L773). It does not appear that this is happening with [the new `spreg` schema](https://github.com/pysal/spreg/pull/27/files#diff-354f30a63fb0907d4ad57269548329e3R48).

**Comment by jGaboardi** [comment]
Let's try to revert back to the original schema?

```
else echo 'testing git libpysal'; git clone 
https://github.com/pysal/libpysal.git; cd libpysal; pip install .; cd ../;
fi;

**Comment by jGaboardi** [comment]
@weikang9009 the [Allowed Failures in Build 82 are now passing](https://travis-ci.org/pysal/spreg/builds/617219282?utm_medium=notification&utm_source=github_status) (where `PYSAL_PYPI=false`). Once a new release of `libpysal` is available via `pip`/`conda`, travis will go green again (with the current [`env matrix`](https://github.com/pysal/spreg/pull/27/files#diff-354f30a63fb0907d4ad57269548329e3R12-R28)).

**Comment by weikang9009** [comment]
@jGaboardi Wow! This is great! Thanks a lot.

 Do we need [`pip install -e . --no-deps --force-reinstall`](https://github.com/pysal/spreg/pull/27/commits/0af6eb609f0d3b85a9bd18ec2e84285bf15fd5b1#diff-354f30a63fb0907d4ad57269548329e3R44)? Seems to be working fine without it.

I also need to adjust the scheme in `giddy` and pin @slumnitz about this.

**Comment by jGaboardi** [comment]
@weikang9009 I thought that it was passing because of the addition of `pip install -e . --no-deps --force-reinstall`, but it seems that I was wrong. Let's try without it to make sure it still the "Allowed Failures" still pass.

**Pull request merged**